File: 2026-08-14-fsharp-spreadsheet/tests/TestHelper.fs
module Spreadsheet.Tests.TestHelper

let mutable Passed = 0
let mutable Failed = 0

let assertEqual (expected: 'a) (actual: 'a) (message: string) =
    if expected = actual then
        Passed <- Passed + 1
    else
        Failed <- Failed + 1
        printfn "FAIL: %s\n  expected: %A\n  actual:   %A" message expected actual

let assertTrue (condition: bool) (message: string) =
    if condition then
        Passed <- Passed + 1
    else
        Failed <- Failed + 1
        printfn "FAIL: %s" message

let assertFalse (condition: bool) (message: string) = assertTrue (not condition) message

File: 2026-08-21-ruby-network-stack/test/test_helper.rb
require 'stringio'
require_relative '../lib/network_stack'

$failures = []
$assertions = 0

def assert(condition, message = 'assertion failed')
  $assertions += 1
  raise message unless condition
end

def assert_equal(expected, actual, message = nil)
  assert(expected == actual, message || "expected #{expected.inspect}, got #{actual.inspect}")
end

def test(name)
  yield
  puts "  ok  #{name}"
rescue StandardError, RuntimeError => e
  $failures << [name, e]
  puts "FAIL  #{name}: #{e.message}"
end
